### PR TITLE
Mixing Hyper-Noblium is no longer required for T4 Lasers, and is now a way to get them for free instead.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## new info as of 4/4/2022
According to a stat dump from oranges, the Quantum Electromagnetic Technology tech node which requires the 16 moles of hypernob to research has been researched only 84 times since 3/3/2022.

In the previous month, it had been unlocked 973 times.
## About The Pull Request

The Hyper-Noblium Gas Shell experiment is now no longer required for Quad Ultra Micro Lasers, and now provides a 100% discount on the node if you do it.

## Why It's Good For The Game

So, I've been telling players complaining about this for *weeks* to just learn how to mix gases, because toxins hasn't had anything for 20 years other than "make bomb".

I figured I'd put my money where my mouth is and try, as a brand new to atmos player, mix hypernob as a Research Director on a private server.

I failed all three attempts I made.
## Attempt #1: "instructions scribbled on back of napkin"
I was told you mix 3 cans of oxygen and 1 can of plasma and light it on fire to get tritium.
This gave me 2 to 3 moles of it, which isn't enough to mix Hyper-nob; you need 5 moles minimum for 1 mole of hyper-nob.
I forgot to take pictures.
## Attempt #2: Guided by Blase
I was walked through using the ordnance computer, and shown the recipe for hyper-nob that way, walking through the many UI issues with the computer.
I was instructed to make a bunch of alterations to set up a gas mixer and such, and properly pump in a 97:3 ratio of oxygen:plasma for tritium. This made enugh tritium.
When I attempted to put it in the can, the can began making acid noises and then exploded, fucking up my workplace irreparably without an atmos tech coming by and fixing it.
dreamseeker_uNn9SNKkD0
Failure. Did as told by an expert, forgot to upgrade the can which isn't communicated to players in any sane way, can exploded from only a few moles of tritium.
## Attempt #3: Guided by Lak
I was walked through the same setup as before, but with additional scrubbers and loading exactly 3 cans in of oxygen and cycling through them, along with how to upgrade my canister through an unintuitive deconstruct/reconstruct method.
After running through the previous setup, after some fiddling with the freezers, we finally produced hyper-nob!
dreamseeker_yR36o5YlWa
An entire 2.6 moles of it.
You need **16 moles** for the research.
We burned through 3 of our 6 cans of oxygen in storage.
dreamseeker_Jxvl4bVD6q
In order to get all 16 moles, according to my math we'd need about 18-19 cans of oxygen. This would require stealing from atmos/stealing directly from the distro.
I'm deeming it a failure because it would require theft to accomplish, and some advanced thinking which is beyond what we expect from both new players approaching a job and "someone walking in to do station infrastructure support", which Toxins now classifies as since it's required for important parts now.

## The verdict:
This isn't doable by a newbie, and certainly isn't doable for actual brand new players, let alone people who are just expected to do the bare minimum.

The default Supermatter setup on the station will run the engine with no fiddling without blowing up, because people need power to do their job as it sucks to work without power.

The default setup for a map should be the bare minimum needed to do the job for everyone else to benefit. Now that mixing Hyper-Noblium is expected as the bare minimum for scientists/RDs because the better parts are locked behind it, the default setup is no longer enough. Toxins is no longer a job you can fuck around with, it's directly required for station progression now.

This is not good game design. I understand why they wanted there to be interdepartmental interaction on this one, but this just sticks everyone else's enjoyment behind a very specific few science players with the right mentality to actually get this done, and nobody is doing this experiment because it's such a pain in the ass.

Is there a way around this? Maybe. But this isn't great right now, and this absolutely can't be something the station needs for progression, because frankly it's not getting done despite the protests of the atmos players in our community who insist they personally do it so therefore everyone else does it in their mind because they lack the ability to place themselves in someone else's shoes who doesn't understand atmos as well as they do.

I've been telling players for weeks to just learn how to mix hypernob.

I gave it the honest college try three times in a row, from starting off from scratch to multiple attempts with atmos professionals giving me a step-by-step on how to do it successfully, and had absolutely no luck reaching that goal of 16 moles of Hyper-Nob for the research to complete.

This should be done by atmos techs, not scientists. They have the time, the knowhow, and the inclination to do it. Not Science players.

Because everyone wants stock parts upgrades, when this doesn't get done, people get hostile and angry at Scientists and RDs who haven't done it. This makes it really unfun to be a Scientist or RD who can't wrap their head around toxins, that job part you could previously ignore because it was only used for making griefing tools as a non-antag and game winning tools as an antag.

## Problem points I discovered
problem point 1:
toxins players should not have to learn the air alarm interface to do their job
problem point 2:
the pipes going in/out should be red/blue to match our color coding elsewhere, ie. plumbing
problem point 3: 
cant do shit w/o t3 lasers
problem point 5: (i forgot 4)
metastation's toxins is broken, the freezers don't work because there's nothing in waste roundstart, cryogenics has the same problem and so does other maps
problem 7: 
no visual way to tell what layer a pipe is on on machinery pipe inputs
problem point 8: 
make the toxins computer window bigger by default
problem point 9: 
different maps might change the toxins todo list
problem point 10:
the ordnance menu has a search bar instead of buttons or a dropdown to select reactions; it should have both if no reactions are selected
problem point 11:
typing in hypernob doesn't give me the hypernob reaction, add tags to this
problem point 12:
the recipe for making tritium is not listed when you type tritium; it's hidden under "plasma combustion"
problem point 14:
tritium should be 3 oxygen to 1 plasma ratio, 97% is such a weird arbitrary number
problem point 15:
"It's relationship with oxygen also determines reaction speed" on the ordnance computer UI doesn't actually explain what this means, had to ask someone for this
problem point 16:
the default pipe layout on the map has to be adjusted for the required job of making tritium, which is expected of all toxins players now due to the experisci changes
the default map layouts should allow the bare minimum of the job to be done; it can be improved by players looking to improve the setup but the bare minimum should work, ala the default Supermatter piping 
problem point 17:
RPD starts at the wrong color, should start at red/blue/black/brown
problem point 19: 
label what direction node 1/2 are relative to the current facing of the mixer, this is a problem with plumbing machinery too
problem 20:
have to modify the internals of the reaction chamber to do my job to bare minimum as toxins worker
problem 21:
airlock control buttons keep reopening when i walk by them
problem 22:
airlock cycling should be instant if the air between the two are identical
problem 23: 
there's pipe colors that are too similar, i just got punked between brown/pale orange, please make the colors more vibrant
problem 24: 
scrubber needs set to expanded range instead of normal range despite being in a 2x3 room
problem point 26: 
can only throw pipes 1 tile???
problem 27: 
one can of oxygen and 1 can of plasma at a 97:3 ratio only made 4.73 moles of trit; need 5 moles for hypernob
problem point 28: 
the concept of upgrading canisters is not explained anywhere; above suggestion about unfinished can w/ plasteel next to it would teach players how to do it naturally, good environmental teaching
problem point 29: 
nowhere in the atmos guides in game does it tell players that moles increase size the hotter they are/decrease size the colder they are, a player brand new to thermodynamics would be confused by this like I was originally years ago
problem point 30: 
cant upgrade a canister, have to build new one/decon old canister frame to rebuild it, suggest adding a rmb tooltip ala plants w/ welder to show decon for that
problem point 31: 
need to be spaceworthy to do initial job setup; this stuff should be setup roundstart
problem point 32: 
analyzer should pop up auto-updating tgui inputs for pipes so i dont have to spam click
## Changelog
:cl:
balance: Mixing Hyper-Noblium is no longer required for T4 Lasers, and is now a way to get them for free instead.
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
Please take comments like that to the forums. Players who've never touched atmos before should be able to do the hypernob experiment if you want it to be required for station infrastructure upgrades.

**Comment by thgvr** [comment]
Goofball finally learns the only way to play atmos is by codediving or begging someone who codedived to help.

**Comment by CMDR-Gungnir** [comment]
Ghilker this is needed to get t4 lasers which everyone is going to bitch for.

This means an RD player needs to be able to do Toxins.

This means you're telling the RD that he has to know the entirety of Science, and the harder half of Engineering, in order to be an RD.

Surely you can see why that's not very smart, right?

**Comment by CMDR-Gungnir** [comment]
> > Ghilker this is needed to get t4 lasers which everyone is going to bitch for.
> > This means an RD player needs to be able to do Toxins.
> > This means you're telling the RD that he has to know the entirety of Science, and the harder half of Engineering, in order to be an RD.
> > Surely you can see why that's not very smart, right?
> 
> Sometimes you don't get everything handed to you, encouraging people to learn new aspects of the game to help better themselves (and the station) in the round is a good thing

Sure. They should do that. By playing Atmos. Not for the bare minimum of a job in AN ENTIRELY DIFFERENT DEPARTMENT.

**Comment by thgvr** [comment]
Update the wiki then, there's already enough secret atmos stuff that isn't available to normal players.

**Comment by SuperNovaa41** [comment]
> 

? the entire focus that we've been moving towards is inter departmental interaction. this is a bad take

**Comment by thgvr** [comment]
@SuperNovaa41 #63033

**Comment by CMDR-Gungnir** [comment]
> > > > Ghilker this is needed to get t4 lasers which everyone is going to bitch for.
> > > > This means an RD player needs to be able to do Toxins.
> > > > This means you're telling the RD that he has to know the entirety of Science, and the harder half of Engineering, in order to be an RD.
> > > > Surely you can see why that's not very smart, right?
> > > 
> > > 
> > > Sometimes you don't get everything handed to you, encouraging people to learn new aspects of the game to help better themselves (and the station) in the round is a good thing
> > 
> > 
> > Sure. They should do that. By playing Atmos. Not for the bare minimum of a job in AN ENTIRELY DIFFERENT DEPARTMENT.
> 
> ? the entire focus that we've been moving towards is inter departmental interaction. this is a bad take

Sure, but that should be in the forms that make sense. Medical should do Dissections because that research benefits them. T4 Lasers benefits the entire god damn station, except the Atmos Wizards who aren't really going to care about it either way...IE the people who you're asking to inter-department with.

And again. If you want that to be the case, then move Toxins to Atmos. Don't make it a requirement for an RD to know, because again, now you have a single Head of Staff who requires the knowledge of his entire department and someone else's, and nobody else needs that.

**Comment by GuillaumePrata** [comment]
People have been blowing/burning themselves, and half of science, for years without anyone complaining about ~~toxins~~ Ordnance being an Engineering job inside of Science and thus it shouldn't be challenging. That is an odd point.

On another note, Atmos is such a bitch against new players that it is hard to get feedback from people that are starting into it, either because they assume it is a skill issue that they will overcome or they give up and don't care.
This PR is at least a gold mine with some good feedback so thanks for that Goof.

**Comment by CMDR-Gungnir** [comment]
That would be because previously it was only used for Advanced But Unnecessary things, like weapons (or free research points/Sci Budget), whereas now it's used for something that people demand (T4 Lasers).

**Comment by Ghilker** [comment]
> Ghilker this is needed to get t4 lasers which everyone is going to bitch for.
> 
> This means an RD player needs to be able to do Toxins.
> 
> This means you're telling the RD that he has to know the entirety of Science, and the harder half of Engineering, in order to be an RD.
> 
> Surely you can see why that's not very smart, right? A Chief Engineer doesn't need to know Medical. The CMO doesn't have to be a master at playing Security. The HoP doesn't even need to be able to do his own department at all, let alone someone else's. This would make Research Director the ONLY Head of Staff that requires the knowledge of two entire departments to be entry-level in.

My point here is that toxins player already have the knowledge to do this, they already know how to make tritium in high amounts. They just need to add a few more knowledge on their curriculum.

**Comment by Iamgoofball** [comment]
i contend that that knowledge is so far out of scope for most of these players that they will just hit their head against the obtuse brick wall that is the ordnance computer while insane people break down their department doors to kill them for not doing their job fast enough, especially if that godawful tools PR gets merged

**Comment by ZerVexon** [comment]
There seems to be a lot of talking past each other here. Yes, Ordnance is not supposed to be easy, but there's something to be said when the main opinion for atmos by people who don't do it is to call it an eldritch magic. I do atmos, it's not hard for me, but I understand where people come from when they say they have no idea how to make Hyper Noblium, or even something simpler, like BZ. At least for the downstream I play on, which is geared towards bringing new people in, the main attractions for new players are medbay and science, and most players barely understand the air distro across the station. Atmos as it exists has a lot of quirks and unintuitive methods that make it hard to approach. 

I can guarantee that some toxin players literally knew to mix plasma and oxygen at 97% because they were told that was correct: not understanding **why** that was correct, just that it was. As much as I enjoy inter-department cooperation, this is not asking for cooperation. There's not enough O2 in Ordnance to do this, you either need to ask for a very full t2 or a t3 can of O2 from atmos, or just ask an atmos tech to make Hybernoblium for you. 

People who want to learn atmos should have a way to start, and previously, I'd say that Toxins was the baby atmos. You didnt get asked to make crazy shit, and you start with one of the most basic reactions: Tritium producing plasma fires. You have space to mess around with pipes and a couple gasses, and handle a very basic reaction. It was a way to learn about pipes without getting overwhelmed by the massive Atmos pipenet. But since you're now asking Ordnance to do the complicated reactions instead, you've taken away that "Baby's first Atmos" course by replacing it with a reputation for being for advanced atmos engineers having something to do with these gasses outside SM interactions.

**Comment by Muffindrake** [comment]
Please merge. The experiment completions in the past weeks have been most likely [citation needed] been attributed to me, myself, and I, which sums up to the estimate of three experiment completions that I gave in another ShItTiFY sTaTiOn PrOgReSsIoN PR, over the couple of weeks. I mostly play atmos tech, and never play toxins, and knowing how to make HN, it is an absolutely ludicrous proposition to ask 'follow a guide' players to perform requisite steps with the small amount of gas present in toxins.

I have a couple of alternate suggestions:

- Decrease heat exchange pipe inefficiency from +20K to +0K, allowing HN formation without the use of thermomachines through heat pipes in space
- Increase maximum temperature for HN formation to 50K. This would allow players to either use heat pipes in space or T2 component thermomachines to get down to the necessary temperature. Currently, you require T3 lasers (and probably T3 matter bins, because slow). Increasing it to 55K or so would allow setups with condenser assemblies. Would have to adjust Halon formation temperature by the same amount, or change the recipe of that (honestly I see no reason to waste tritium on that gas)
- Lock T4 lasers behind an experiment that involves rolling the dice and choosing a head of staff pet to throw into the supermatter

One of these suggestions is less serious than the other two.

**Comment by Ghilker** [comment]
Saying that toxins is just baby atmos is the worst thing you could do to toxins, if it was it wouldn't make sense for it to exist at all. Toxins should be it's own department that does gas sperimentation and observation but you are treating it as if its just a passage point towards atmos and it shows in how much nobody cares about toxins. 
All this does is make toxins easier by dumbing it down and it doesn't touch the issue you are all complaining about.
Just remove it already

**Comment by ZerVexon** [comment]
> Toxins should be it's own department that does gas sperimentation and observation 

Then what, exactly, is atmos for? If you asked "which department should do gas experiments and observations" I would assume more people would say atmos. Toxins was for making bombs using a niche aspect of atmos, which was important for the other point.

> Saying that toxins is just baby atmos is the worst thing you could do to toxins, if it was it wouldn't make sense for it to exist at all. 

No, that's backwards. It makes **more** sense for something to exist this way. I have gone through each and every role on the station over the past two years, and Atmos is by far the most difficult department on the station. You don't start by teaching someone calculus, you start with arithmetic. There's an order to learning things, and having something be a simpler, niche example where you can do more experimentation and learn the basics is how you step towards more advanced processes. 

Toxins worked as a sort of introduction to atmos this way by focusing exclusively on two interactions (Tritium producing plasma fires and explosive combustion), allowing much more freedom and opportunity to experiment with systems you know, since you weren't expected to try and make 20 different gasses. You weren't tasked with much more than "make tritium" and that's literally the first gas interaction for atmos. Everything else is more complicated than that one, even electrolysis to make H2.

**Comment by GuillaumePrata** [comment]
Ordnance being baby Atmos was good because the main Atmos room is a scary thing to work with, I'm pretty sure we all had our share of rounds where an Atmos tech link waste to distro and the shuttle is called 15 minutes in.

It would be better for the game if it was the other way around, Atmos tech is a round maintenance job, they are important to go around fixing station damage or the shuttle gets called on the first bombing but most of the Atmospherics content is about sitting in a room by yourself.

Flowing the hard side of Atmospherics into science (A department that is about doing "hard stuff" to improve the station) and leave Atmos techs to be focused on repairs could be an interesting change.

**Comment by Ghilker** [comment]
> Then what, exactly, is atmos for? If you asked "which department should do gas experiments and observations" I would assume more people would say atmos. Toxins was for making bombs using a niche aspect of atmos, which was important for the other point.

Atmos is for managing the station atmospheric, fighting fires and fixing air issues, while also having the ability to mass produce gases when needed. 

> No, that's backwards. It makes **more** sense for something to exist this way. I have gone through each and every role on the station over the past two years, and Atmos is by far the most difficult department on the station. You don't start by teaching someone calculus, you start with arithmetic. There's an order to learning things, and having something be a simpler, niche example where you can do more experimentation and learn the basics is how you step towards more advanced processes.
> 
> Toxins worked as a sort of introduction to atmos this way by focusing exclusively on two interactions (Tritium producing plasma fires and explosive combustion), allowing much more freedom and opportunity to experiment with systems you know, since you weren't expected to try and make 20 different gasses. You weren't tasked with much more than "make tritium" and that's literally the first gas interaction for atmos. Everything else is more complicated than that one, even electrolysis to make H2.

You start with atmos to learn atmos, not with a totally different department. You have all the tools, immense amount of gases and plenty of room to play with and make big setups without worrying about optimal space. Toxins is hard on space and resources and demands you to learn how to manage those two while creating setup that produces enough to sustain bombs worth doing. How's toxins easy again? Oh yeah copy pasted metà setups done in five minutes without understanding, gotcha.

**Comment by Camahoo** [comment]
> Oh yeah copy pasted metà setups done in five minutes without understanding, gotcha.

...Is that not how a lot of stuff in game is done? Maybe I've missed something, but I think there's more than a few things where someone figured out the best way to do it, put it on the wiki or spread through word of mouth, and then that was the way people did it, without putting too much effort into learning why.

Also, are t4 lasers worth this much trouble? Again, I've hardly memorized the wiki (and the wiki is hardly perfect), maybe I missed it, but there honestly didn't seem to be a plethora of machines that benefitted from them and were actually used often. On top of that, pretty sure t4 lasers are the only t4 part locked behind an experiment right now, which only seems strange given that I honestly found all of two machines that might be a priority to upgrade with them.

Granted, I'm downstream right now, so maybe a whole bunch of experiments have been added that I just hadn't seen yet.

**Comment by GuillaumePrata** [comment]
> ...Is that not how a lot of stuff in game is done?

One issue is that Ordnance has been the same meta rush for so long and now that people need to learn new and different things it became a mess again.
Everyone got used with Ordnance being something that is being speedrun every shift with no challenge, now that content is being added into it and that people are failing to unlock things it sparked all this conversation around what is fair to be behind the wall.

**Comment by CRITAWAKETS** [comment]
people are making meta setups right now with ordnance and so far it seems to be the only way that content is even done

**Comment by LemonInTheDark** [comment]
Before I get too far into this, I wanted to go over my general thoughts/approach to toxins and atmos as it currently is.

## Atmos as it is

I think atmos as a section of the game has a serious issue with penetrability. Many things are harder then they ought to be, and many semi simple mechanics are quite poorly explained.
I want to fix this, because I think it presents a real issue for having atmos tech the job actually do anything productive or engage with the round by default (See rant about broom closets and lack of human contact) 

Vin especially has been trying to fix the part of this. Even having the reactions in the game is a large step forward.
The problem is atmos tech doesn't have a lot of reason to like, do anything. Most of its content is optional, and it selects for people like our atmos coders, so if there's an issue with learning it, we don't really find out. 

I'd like to use toxins and gas experiments as a testing ground for atmos as it stands, and to test if it's workable to learn.
I don't have any particular quibble about what is locked off, but I do want it to be something you notice being gone. 
I think? T4 parts accomplish this reasonably well, I don't think they're exactly mandatory for anything outside of some atmos memes, but you still notice if they're not there.

I realize this is slightly unfair to current toxin players, but I can't think of a better way to institutionally unfuck things.
I do think the scaling of difficulty on toxin experiments is a bit wack, I've talked to vin a bit about that and I'm hopeful it'll get a bit better. I to some extent trust him to make intelligent decisions on the topic, since he's spent a lot of time working on it, and appears to care about making his content more accessible.
The test thing wasn't his idea mind, that's me being evil not anyone else. Insert disney monologue here,

Anyway, I wanted to go over some of the painpoints you ran into, if that's chill.

## Stuff that stood out to me

> When I attempted to put it in the can, the can began making acid noises and then exploded, fucking up my workplace irreparably without an atmos tech coming by and fixing it.

If memory serves, can tiers was added to lessen 50K rushing, and to some extent to handle can fusion memes, 
I'm considering removing it, since those problems are behind it, and it's an amount of extra brain weight that I don't think? needs to exist.

> In order to get all 16 moles, according to my math we'd need about 18-19 cans of oxygen. This would require stealing from atmos/stealing directly from the distro.

That seems quite a bit off. Now I don't actually play the game mind, but hypernob takes 5 moles of trit for one mole, something's really off if a canister of o2 (1800 moles) gets you less then 5 mols of trit. At least I think, I assume you'd need more then that to make a 50k bomb.

> Because everyone wants stock parts upgrades, when this doesn't get done, people get hostile and angry at Scientists and RDs who haven't done it. This makes it really unfun to be a Scientist or RD who can't wrap their head around toxins, that job part you could previously ignore because it was only used for making griefing tools as a non-antag and game winning tools as an antag.

This sucks yeah. To some extent it's gonna happen so long as we make any aspect of atmos required, but people getting bullied for it is pain.

## Wew lord

> 1: toxins players should not have to learn the air alarm interface to do their job 

I'm not so sure about this. I think they already did need to learn it to make bombs, and not knowing it is a major blocker to doing anything with atmos, which I do want them to be doing.

> 2: the pipes going in/out should be red/blue to match our color coding elsewhere, ie. plumbing 

This causes problems because pipe colors are used for connecting pipes. It's very important that toxins doesn't accidentally connect the air network to its tritium fire. For obvious reasons.

> 3: cant do shit w/o t3 lasers 

Hmmf that's not the best. It's the result of trying to gate hypernob as a higher tier atmos gas, but it does limit people somewhat. Maybe it would help if you had a reason to make a freezer before producing HN so it could just be upgraded?

> 5: (i forgot 4) (lmao) metastation's toxins is broken, the freezers don't work because there's nothing in waste roundstart, cryogenics has the same problem and so does other maps problem 

It's not broken persay. 
That's the result of something ghil was trying to do with heat management that I shut him down on because of the added seemingly (to me) pointless complexity. I've been avoiding handling it because I don't like arguing with my coders and it wasn't actively breaking anything, maybe I need to spend some social capital 

> 7: no visual way to tell what layer a pipe is on on machinery pipe inputs 

You sure? that seems a bit weird. You mean like freezers right?

> 8: make the toxins computer window bigger by default 

This'd be fine

> 9: different maps might change the toxins todo list 

In relation to collecting o2? I'm ok with this in general, but it depends on how disruptive it is. Variety is useful, but atmos is hard to learn so there do need to be some consistent parts.

> 10: the ordnance menu has a search bar instead of buttons or a dropdown to select reactions; it should have both if no reactions are selected 

It should have a button that lets you view all reactions and gasses in a big list you can hop through. Best case scenario with pictures.

> 11: typing in hypernob doesn't give me the hypernob reaction, add tags to this 

Yeah. Rather then tags, we should maybe support searching more fields then the id.

> 12: the recipe for making tritium is not listed when you type tritium; it's hidden under "plasma combustion" 

See above

> 14: tritium should be 3 oxygen to 1 plasma ratio, 97% is such a weird arbitrary number 

I don't know about this. 75% will work, it's just not the optimal mix, which I think is ok. Maybe the trouble is more making that information clear.

> 15: "It's relationship with oxygen also determines reaction speed" on the ordnance computer UI doesn't actually explain what this means, had to ask someone for this 

Should be more clear yes

> 16: the default pipe layout on the map has to be adjusted for the required job of making tritium, which is expected of all toxins players now due to the experisci changes the default map layouts should allow the bare minimum of the job to be done; it can be improved by players looking to improve the setup but the bare minimum should work, ala the default Supermatter piping 

I don't think I agree with this. I know vin is working on some remaps of toxins to make it easier to use https://github.com/tgstation/tgstation/pull/65807
Nobel goals. I guess I think of this reaction should be more then the first one you do, and an important skill/part of the fun of atmos is designing pipes. The way vin has his current remaps setup is built for bz production, which I'm happy with as a starting reaction. I think this somewhat fits what you're talking about.

> 17: RPD starts at the wrong color, should start at red/blue/black/brown

Ehhhhhhhhhh it varies based off the map and rpd placement and and.
I'm ok with green as a default, you're going to need to switch colors at some point regardless
 
> 19: label what direction node 1/2 are relative to the current facing of the mixer, this is a problem with plumbing machinery too 

It might be better to just say their position relative to the output. Agree

> 20: have to modify the internals of the reaction chamber to do my job to bare minimum as toxins worker 

See problem 16, though it does depend on the severity of the internal change.

Aside: 
~~keep your syntax the same so I can easily regex the problem entries hhhhhhh~~

> 21: airlock control buttons keep reopening when i walk by them 

That sounds strange, what do you mean?

> 22: airlock cycling should be instant if the air between the two are identical 

I don't see any reason this can't be the case, yeah.

> 23: there's pipe colors that are too similar, i just got punked between brown/pale orange, please make the colors more vibrant 

You have any pictures, so it's a bit hard to agree or disagree with you.
< doesn't play the game

> 24: scrubber needs set to expanded range instead of normal range despite being in a 2x3 room 

This is so you can directly harvest the gas reactions on the two tiles the scrubber can get to, rather then just its own.

Haha forgot 25 point and laugh

> 26: can only throw pipes 1 tile???  

The fuck

> 27: one can of oxygen and 1 can of plasma at a 97:3 ratio only made 4.73 moles of trit; need 5 moles for hypernob 

This seems really off, maybe I'm dumb? See above

> 28: the concept of upgrading canisters is not explained anywhere; above suggestion about unfinished can w/ plasteel next to it would teach players how to do it naturally, good environmental teaching

See also above, I'm not sure it needs to exist anymore. If it does yes this is a good point.
 
> 29: nowhere in the atmos guides in game does it tell players that moles increase size the hotter they are/decrease size the colder they are, a player brand new to thermodynamics would be confused by this like I was originally years ago 

We need an intro page yeah.

> 30: cant upgrade a canister, have to build new one/decon old canister frame to rebuild it, suggest adding a rmb tooltip ala plants w/ welder to show decon for that 

See above above above.

> 31: need to be spaceworthy to do initial job setup; this stuff should be setup roundstart 

Is this about getting cooling setup?

> 32: analyzer should pop up auto-updating tgui inputs for pipes so i dont have to spam click

I'm not sure about this. I don't want to make it only work by ui open. 


## Lemon you motherfucker give me a TL;DR

I'm evil, and want to use toxins as a way to make atmos not fucking hell to interact with. Part of that is locking something meaningful with it. It makes me sad that this is driving people out of sci, if that is happening.
HN shouldn't be as o2 consuming as you described, that's weird.
Atmos has a lot of things that are more tedious or complex then they need to be. This will be fixed.
Atmos has a few concepts that are quite poorly explained in game. This should be resolved.
Goof forgot problem number 25 point and laugh.
I want HN production to be a harder step in a gradient of difficulty in toxins.
I am evil

In light of the above, I don't think I'm gonna merge this. 
I don't like people being driven out of sci, but if we don't force atmos it'll remain as it is now, which is not great.

I am however very interested in the painpoints people have with atmos content. If you have some to share, please do post it here (Why'd you not put it in feedback vin you bastard) 
https://tgstation13.org/phpBB/viewtopic.php?f=83&t=3143

**Comment by CMDR-Gungnir** [comment]
Lemon, in order to be even remotely accepted as a Head of Staff player (and on the more RP-based servers, it's pretty much mandatory) you have to be able to do every part of your department.

Toxins being made into Total Atmos instead of just "Hey I know how to make da big boomies" forces the Research Director to also be an Atmos Main, as well as his own department. This is the only Head of Staff that would be like that. Please explain why you think this is a good idea.

**Comment by RikuTheKiller** [comment]
> ## Lemon you motherfucker give me a TL;DR
> I'm evil, and want to use toxins as a way to make atmos not fucking hell to interact with. Part of that is locking something meaningful with it. It makes me sad that this is driving people out of sci, if that is happening. HN shouldn't be as o2 consuming as you described, that's weird. Atmos has a lot of things that are more tedious or complex then they need to be. This will be fixed. Atmos has a few concepts that are quite poorly explained in game. This should be resolved. Goof forgot problem number 25 point and laugh. I want HN production to be a harder step in a gradient of difficulty in toxins. I am evil
> 
> In light of the above, I don't think I'm gonna merge this. I don't like people being driven out of sci, but if we don't force atmos it'll remain as it is now, which is not great.

Oh, about driving people out of sci, YES, that is happening.

(We've had far more rounds with no scientists and also I've personally quit science as well due to the gas shell experiments, though I've finally bashed my head enough at a brick wall to be able to make hyper-nob.)

Just to reference how bad this is, I've seen t4 lasers done ONCE EVER SINCE the update. For scale, that's like around 4 rounds per day, let's say each over an entire hour and still no more than once have I seen the lasers. Even that time the person wearing an atmos suit literally said "that took way too long", so yeah...

Gas shells, in general, aren't a great way to go about doing this. Toxins has just become a discount atmos that could literally just be moved to atmos now as a seperate department from sci. That's stupid, genuinely. Scientists shouldn't have to learn an entire fucking department for a single research. Like, please. And the fact there are lots of gas shells makes this FAR worse. If it was just one gas shell, sure. Two? Okay, yeah, gives the compressor more than one job... and maybe doable in a single round. Three? Very obnoxious now... Four? Yeah no, all of these are never going to get done in a single round. Ever. That's simply a fact.

**Comment by moocowswag** [comment]
> Oh, about driving people out of sci, YES, that is happening.

Same except I just do circuits in maint now instead of quitting sci entirely.

The vast majority of the tech tree was not designed to be locked behind brick walls and could probably use a rework at this rate.

> Can someone make a guide please preferably in video form.

The first and last time I made a video guide on how to do something toxins related became outdated within like a month which is why I think if someone is gonna make changes like this they need to be the one that is prepared to hold a player`s hand and guide them through their intended way to do a task to unlock a tech.

We wouldnt be in this mess right now if the individual that made t4 lasers require hypernoblium also made a walkthrough stating how many steps scientists had to go through to get their tech and move on with their life.

Stating the process youre expected to go through to fulfill a requirement that your pr is adding should already be the norm. Just pr-ing "Makes hypernoblium a requirement for t4 lasers" is not transparent enough for even the maintainers to make a concious decision on merging or not because most maintainers most likely havent touched toxins in months.

**TLDR if you add hoops for players to jump through you need to communicate to everyone how many hoops are being added and how high they are off of the ground. This not only helps maintainers know what they are merging but also helps the playerbase adapt to the change.**

**Comment by Whoneedspacee** [comment]
shouldve just left atmos point gen as it was but now all anyone does is build 50k bombs because its hard capped and then that got nerfed too

i don't think any atmos player really cares if this specific change is made but peeps gotta stop neutering the department at the same time as admitting that it's very difficult

clearly copy pasting designs isn't as easy as you thought so i just hope you consider that in the future

**Comment by Pandarsenic** [comment]
My favorite part of Toxins is that the _slightest_ fuckup, even if it doesn't burn down the entire lab, generally renders the department _unusable for the remainder of the round_ because of shit in the pipes, messed up burn chamber, lack of the gas you need to complete what you're trying to do, etc.

One person gets one try at Toxins each round and if something goes wrong nobody gets anything from the whole department :)

**Comment by optimumtact** [comment]
every argument here seems to rest on the fact that it's peoples god given right to unlock t4 parts every round.

**Comment by CMDR-Gungnir** [comment]
> every argument here seems to rest on the fact that it's peoples god given right to unlock t4 parts every round.
> 
> i'm perfectly fine with the t4 parts being hard to unlock to be honest.

What you have to consider is that it feels bad to have something you USED to have, get locked behind something new and obnoxious that makes you pretty much never get it. I don't think people would mind as much if T5 parts were added and these experiments were for them. It just sucks to have something taken away.

**Comment by LemonInTheDark** [comment]
It sucks, but that's how it be.
If there's a problem it's with people being bullied over it, but that's a player and admin issue, not an us thing.

**Comment by Whoneedspacee** [comment]
to be frank its not hard to make hypernoblium once you’ve done it once

it might take you 20 rounds maybe more but how many rounds did you spend to down your first solo megafauna

once you’ve got it down you’ve got it down simple as that

**Comment by MetalClone** [comment]
Add tier five abductor parts. Also holy shit atmos players literally going insane over 3 fucking lines of code LOL

**Comment by optimumtact** [comment]
that's a lotta words - shame i'm not gonna be readin' em

**Comment by Draggeru** [comment]
I never thought I'd simp for a goof PR but here we are.

**Comment by OzzyMcPseudonym** [comment]
Honestly atmos being optional as a department is perfectly fine with me. It's more obtuse than chemistry and botany by a mile, doesn't produce anything unique or cool like xenobiology, doesn't give valuable combat experience to new players like mining does. Honestly the only real use of atmos is either griefing via plasmaflood or, if you are one of the atmosian elder gods, getting spaceproof armor and a metallic hydrogen fireaxe that is worse than the one you start with in your department anyway. There's no real incentive to do anything with the atmos system as it currently stands because of several factors

1) Once you know how to do the one meta setup, you don't have to really think about it anymore, it's just rote memory at that point like a chemistry factory.

2) Some of the gasses are so obtuse to make that new people won't be able to get them at all and experienced players will still take a decent chunk of time to get the more exotic gasses in the game. Everyone jokes about atmos players being elder gods because the department as a whole is so obtuse that codediving is practically mandatory if you want to figure something out.

3) Atmos has many quirks that you have to specifically know or it won't work. Stuff like pipenets and how they are formed, stuff like how thermodynamics work in general (as mentioned by Goof in the PR body.)

A subsystem of the game with all of these problems should not be made mandatory in it's current state. If atmos is going to be mandatory, at least make it more accessible to newer players as a whole first. If the system becomes as accessible as something like chemistry or botany on a baseline level then perhaps making atmos mandatory for progression will be more tenable an idea in general. The pain associated with this change is caused by the difficulty *wall* of learning atmos in the first place.

**Comment by LemonInTheDark** [comment]
> that's a lotta words - shame i'm not gonna be readin' em
> 
> @LemonInTheDark please do something with this PR already.

Yes boss

I'm gonna close this. I don't want to remove this pressure from solving atmos's issues, because if I do they will never be resolved.
I'd in theory be ok with making the requirements more stringent, but there needs to be something that serves this purpose.

I realize this causes player interaction problems, but those will hopefully be temporary, and I can't think of another way to handle this properly. Worst case, stop being such assholes to each other.

Godspeed lads.

**Comment by carshalash** [comment]
image
